### PR TITLE
User Folder

### PR DESCRIPTION
**Pull request by DracoClaw**
This PR moves the plugin, theme, and settings folders into a user subfolder. helping to better organize the structure, and clearly indicate that those are the only things end users should be touching.

### CODE REVIEW

**Comment by asportnoy** [comment]
Please update the old migration to move the files to the new folder. This way we don't have to move them twice.

File: .gitignore [hunk 1]
```diff
@@ -7,12 +7,12 @@ package-lock.json
 yarn.lock
 pnpm-lock.yaml
 
+user/*
 
-plugins/*
 !plugins/.gitkeep
```
Comment (asportnoy) [inline .gitignore:12]:
```suggestion
user/*
!user/*/.gitkeep
```
Also make a settings gitkeep while you're at it

File: .gitignore [hunk 2]
```diff
@@ -7,12 +7,12 @@ package-lock.json
...
 
 src/Powercord/coremods/react-devtools/ext
-themes/*
+
 !themes/.gitkeep
```
Comment (asportnoy) [inline .gitignore:16]:
```suggestion
```

File: src/fake_node_modules/powercord/constants.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ module.exports = Object.freeze({
   REPO_URL: 'replugged-org/replugged',
 
   // Runtime
-  SETTINGS_FOLDER: join(__dirname, '..', '..', '..', 'settings'),
+  SETTINGS_FOLDER: join(__dirname, '..', '..', '..', 'user', 'settings'),
```
Comment (asportnoy) [inline src/fake_node_modules/powercord/constants.js:10]: Let's set up a const for plugins and themes and use that

**Review by asportnoy — CHANGES_REQUESTED** [review]

**Comment by lexisother** [comment]
The migration script in the state it's in now won't work for people who have already migrated to the new directories.

**Comment by asportnoy** [comment]
There are 2 scripts

**Comment by lexisother** [comment]
Why not update the old one?

**Comment by asportnoy** [comment]
The old one is for src/Powercord/x to user/x
The new one is for x to user/x

**Comment by lexisother** [comment]
Right.

**Review by lleyton — APPROVED** [review]
hmmmm, something about modifying past migrations, especially if they've already been deployed, just really doesn't seem right (and there's a new migration anyways, so is it necessary?)

regardless, everything else seems okay, but I want someone to locally test this before merging

**Review by lleyton — CHANGES_REQUESTED** [review]
users seem to be having issues with this
https://discord.com/channels/1000926524452647132/1000955965304221728/1012098224930168892

**Comment by bop34** [comment]
Replugged doesn't work at all for me with this branch. It loads when starting Discord, but then returns
`An error occurred while initializing "settings.js"! Error: ENOENT: no such file or directory, mkdir 'C:\Users\[user]\powercord\user\settings'
    at mkdirSync (fs.js:987:3)
    at e.mkdirSync (electron/js2c/asar_bundle.js:5:12191)
    at Object.<anonymous> (C:\Users\[user]\powercord\src\Powercord\apis\settingsStore\store.js:8:3)
    at Object.<anonymous> (C:\Users\[user]\powercord\src\Powercord\apis\settingsStore\store.js:118:3)
    at Module._compile (internal/modules/cjs/loader.js:1078:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1108:10)
    at Module.load (internal/modules/cjs/loader.js:935:32)
    at Module._load (internal/modules/cjs/loader.js:776:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
    at Function.o._load (electron/js2c/renderer_init.js:33:379)`
and
`Uncaught (in promise) TypeError: Cannot read property 'buildCategoryObject' of undefined
    at Powercord.startup (C:\Users\[user]\powercord\src\Powercord\index.js:119:44)
    at async Powercord.init (C:\Users\[user]\powercord\src\Powercord\index.js:90:5)`

**Comment by lexisother** [comment]
That's because the directory isn't being created recursively. `user` does not exist so it cannot create `settings`.

**Comment by asportnoy** [comment]
> hmmmm, something about modifying past migrations, especially if they've already been deployed, just really doesn't seem right (and there's a new migration anyways, so is it necessary?)
> 
> regardless, everything else seems okay, but I want someone to locally test this before merging

I think in this case, it's alright. But if we wanted to change it, we could either:
- Rename the original one to make it a "new" migration
- Merge it into the new migration

File: .gitignore (hunk 1 above)
Comment (asportnoy) [reply .gitignore]: @DracoClaw doesn't look like the gitkeep files were actually commited, can you double check?

Comment (DracoClaw) [reply .gitignore]: should be there now

**Comment by lleyton** [comment]
@asportnoy I think the option of renaming the original to make it into a "new" migration would work

File: src/Powercord/migrations/20220824-move-user-items.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+const { join } = require('path');
+
+module.exports = () => {
+  const oldPlugs = join(__dirname, '..', '..', '..', 'plugins');
+  const newPlugs = join(__dirname, '..', '..', '..', 'user', 'plugins');
+  const oldThemes = join(__dirname, '..', '..', '..', 'themes');
+  const newThemes = join(__dirname, '..', '..', '..', 'user', 'themes');
+  const oldSettings = join(__dirname, '..', '..', '..', 'settings');
+  const newSettings = join(__dirname, '..', '..', '..', 'user', 'settings');
```
Comment (asportnoy) [inline src/Powercord/migrations/20220824-move-user-items.js:10]: Use the constants for the new directories

**Review by asportnoy — CHANGES_REQUESTED** [review]
Make the quick css changes as discussed in Discord

**Comment by asportnoy** [comment]
Closing per poll in Discord

**Pull request closed without merging**